src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2220 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 20 Feb 2024
+
+Copyright © 2024 Delmic
+
+This file is part of Odemis.
+
+Delmic Acquisition Software is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License as published by the Free Software Foundation, either version 2 of the License, or (at your
+option) any later version.
+
+Delmic Acquisition Software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+more details.
+
+You should have received a copy of the GNU General Public License along with Delmic Acquisition Software. If not, see
+http://www.gnu.org/licenses/.
+"""
+import logging
+import math
+import os
+import queue
+import re
+import threading
+import time
+import zipfile
+from concurrent.futures import CancelledError
+from typing import Optional, Union
+
+import msgpack
+import msgpack_numpy
+import notify2
+import numpy
+import Pyro5.api
+import pkg_resources
+from Pyro5.errors import CommunicationError
+
+from odemis import model
+from odemis import util
+from odemis.model import (CancellableFuture, CancellableThreadPoolExecutor,
+                          DataArray, HwError, ProgressiveFuture,
+                          StringEnumerated, isasync)
+
+Pyro5.api.config.SERIALIZER = 'msgpack'
+msgpack_numpy.patch()
+
+# Acquisition control messages
+GEN_START = "S"  # Start acquisition
+GEN_STOP = "E"  # Don't acquire image anymore
+GEN_TERM = "T"  # Stop the generator
+
+# List of known supported resolutions.
+# Although the API only provide the min/max of resolution in X/Y, not every value
+# or combination works. Actually, setting X always changes Y to the corresponding
+# value. Note that they are not all the same aspect ratio. "Legacy" resolutions
+# are ~8/7 while the new ones are 3/2. The "legacy" resolutions end up with a
+# larger vertical field of view.
+RESOLUTIONS = (
+    (192, 128),
+    (512, 442),
+    (1024, 884),
+    (2048, 1768),
+    (4096, 3536),
+    (768, 512),
+    (1536, 1024),
+    (3072, 2048),
+    (6144, 4096),
+)
+
+# Xtadapter debian package installation directory which contains xtadapter's zip files
+XT_INSTALL_DIR = "/usr/share/xtadapter"
+
+
+class Package(object):
```

Comment:
This class and `check_latest_package` can be imported from [here in xt_client](https://github.com/delmic/odemis/blob/2419a9a7ec2f2e4ae309cdbc0d83c529bc958f0b/src/odemis/driver/xt_client.py#L82)

Is there a common place or can you recommend a place where these can go @pieleric ?

Reply:
Ahh I see it is not used